packages/sanity/src/core/validation/types.ts

```diff
@@ -0,0 +1,23 @@
+import type {LocaleSource} from '../i18n'
+
+declare module '@sanity/types' {
+  /**
+   * Extended validation context that includes internationalization
+   *
+   * Why is this not directly part of `@sanity/types`, you ask?
+   * Because `@sanity/types` shouldn't need to depend on the `i18next` package, which it needs
+   * for the `TFunction` type. The `ValidationContext` should never have been part of the types
+   * module in the first place, but is now unfortunately part of the public API and thus cannot
+   * be changed easily.
+   *
+   * This is a temporary solution until we can remove the `ValidationContext` from the types module,
+   * which is likely to happen at the next major version.
+   *
+   * @public
+   */
+  interface ValidationContext {
+    i18n: LocaleSource
```

Comment:
If I'm not mistaken, this will probably appear as a property in the rendering of the docs ui, we might want to just add a small comment / linking it back to the LocaleSource type

Reply:
Will add after merging, thanks for pointing that out